extensions/src/main/java/org/pentaho/platform/plugin/action/olap/impl/OlapServiceImpl.java

```diff
@@ -504,6 +504,12 @@ public void flushAll( IPentahoSession session ) {
       resetCache( session );
 
       flushHostedAndRemote( session );
+
+      MondrianServer mondrianServer =
```

Comment:
You will need to clear both server instances. The bug states that there are inconsistent methods for actually acquiring a mondrian instance used by the various tools. Your patch here does not address that, it only implements the workaround outlined in the case.

As there are still multiple instances alive, you have to call the flushSchemaCache() line on both the active mondrian server (this.server) and the one with Id "null". You cant just clear one, but not the other.